### PR TITLE
feat(DataList): add selectable variant

### PR DESCRIPTION
**Pull request by nicolethoen**
addresses #3233 

I have not figured out a clean way to prevent clicking the kebab (or interacting with components which have their own event handlers) from also selecting a DataListItem row. Events keep propagating up to parent components. I'd love any help people are willing to give... (@evwilkin @jschuler).

Is it possible the interaction as is would be acceptable for merging before code freeze monday? I could open an issue to figure out the last part? @mcarrano @mcoker

### CODE REVIEW

**Comment by mcarrano** [comment]
@nicolethoen without actually seeing this, my opinion is that it would be a significant usability problem if opening a kabob, for example also selected the row. If we were to release this, I would only do so with the restriction that the row should not include active elements to avoid this behavior. @LHinson @rachael-phillips do you recall who was asking for this feature and what their intended usage was?

**Comment by dlabaj** [comment]
@nicolethoen Can you do a prevent default when you click the kebab?

File: packages/patternfly-4/react-core/src/components/DataList/DataListItem.tsx [hunk 1]
```diff
@@ -12,32 +14,77 @@ export interface DataListItemProps extends Omit<React.HTMLProps<HTMLLIElement>,
...
-        React.isValidElement(child) &&
-        React.cloneElement(child as React.ReactElement<any>, {
-          rowid: ariaLabelledBy
-        })
-    )}
-  </li>
-);
+export class DataListItem extends React.Component<DataListItemProps> {
+
+  // @ts-ignore
```
Comment (tlabaj) [inline packages/patternfly-4/react-core/src/components/DataList/DataListItem.tsx]: Why do we need this?

Comment (nicolethoen) [reply packages/patternfly-4/react-core/src/components/DataList/DataListItem.tsx]: typescript has some issues with pulling in context this way. For a handful of versions of React, pulling in a static context was a deprecated practice. It is now supported again, but the ts linter doesn't know that? All instances of a static context in the project need the ts-ignore flag. And its the only way to reference the context outside of a context.consumer in a render function.

**Comment by nicolethoen** [comment]
I rebased with the latest changes, but now selecting the rows doesn't appear to be working. let me investigate quickly

**Comment by nicolethoen** [comment]
> I rebased with the latest changes, but now selecting the rows doesn't appear to be working. let me investigate quickly

I take it back! It looks like the example is working in the surge preview :) @mcarrano @mcoker @tlabaj

**Pull request merged**